kong/db/schema/others/declarative_config.lua

```diff
@@ -506,6 +507,71 @@ local function validate_references(self, input)
 end
 
 
+function DeclarativeConfig.validate_references_full(self, input)
+  return validate_references(self, input)
+end
+
+
+function DeclarativeConfig.validate_references_sync(deltas, deltas_map)
+  local errs = {}
+
+  for _, delta in ipairs(deltas) do
+    local item_type = delta.type
+    local item = delta.entity
+    local ws_id = delta.ws_id or kong.default_workspace
+
+    local foreign_refs = foreign_references[item_type]
+
+    if not item or item == ngx.null or not foreign_refs then
+      goto continue
+    end
+
+    for k, v in pairs(item) do
+
+      local foreign_entity = foreign_refs[k]
+
+      if foreign_entity and v ~= null then  -- k is foreign key
+
+        local db = kong.db[foreign_entity]
+
+        -- try to find it in deltas
+        local pks = DeclarativeConfig.pk_string(db.schema, v)
+        local fvalue = deltas_map[pks]
+        if fvalue then
+          goto found
```

Comment:
should be `break`?

Reply:
it should be `continue` logic, but lua does not support it, so i jump to `found` label.

Reply:
it means, for current `item[k]` 's foreign reference, it has found its associated referenced foreign entity for `item[k]`. so we just need to skip this round of checking `item[k]`, trying to check next `k`.

Reply:
could you add a comment to explain this behavior? thanks.

Reply:
```

      -- Try to check if item's some foreign key exists in the deltas or LMDB.
      -- For example, `item[k]` could be `<router_entity>["service"]`, we need                                                                     -- to find the referenced foreign service entity for this router entity.
```

done